src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/flow_units/PublisherEventFlowUnit.java

```diff
@@ -0,0 +1,52 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.Consts.Table;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.deciders.PublisherEvent;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.FlowUnitMessage;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.GenericFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.util.InstanceDetails;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.PersistableObject;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.PersistorBase;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.TableInfo;
+import java.sql.SQLException;
+
+/**
+ * A flowunit that carries PublisherEvent to downstream. The purpose of creating this flowunit is
+ * to leverage the persistor in RCA framework to persist PublisherEvent and its nested actions.
+ * This is because only flowunit can be persisted into SQL in RCA framework.
+ */
+public class PublisherEventFlowUnit extends GenericFlowUnit implements PersistableObject {
```

Comment:
Instead of persisting via the flow units mechanism, let's just use the `ActionListener` to persist them. Then we don't need a separate flowUnit for publisher.